server/src/main/java/org/opensearch/index/engine/DocumentIndexWriter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.engine;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.LiveIndexWriterConfig;
+import org.apache.lucene.index.Term;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Primary interface for document indexing operation in OpenSearch. This interface is mapped after Lucene's IndexWriter.
+ *
+ */
+public interface DocumentIndexWriter extends Closeable {
+
+    long getFlushingBytes();
+
+    long getPendingNumDocs();
+
+    LiveIndexWriterConfig getConfig();
+
+    boolean hasPendingMerges();
+
+    boolean hasUncommittedChanges();
+
+    Throwable getTragicException();
+
```

Comment:
What exception is this supposed to capture?

Reply:
It is mapped after [Lucene's getTragicException() function call](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L5761).